client/web-sveltekit/src/lib/navigation/GlobalSidebarNavigation.svelte

```diff
@@ -0,0 +1,153 @@
+
+    function isNavigationMenu(entry: NavigationEntry | NavigationMenu): entry is NavigationMenu {
+        return entry && 'children' in entry
+    }
+</script>
+
+<script lang="ts">
+    import { mdiClose } from '@mdi/js'
+    import { page } from '$app/stores'
+    import { onClickOutside } from '$lib/dom'
+    import Icon from '$lib/Icon.svelte'
+    import SourcegraphLogo from '$lib/SourcegraphLogo.svelte'
+
+    import { isCurrent, mainNavigation } from './mainNavigation'
+    import MainNavigationLink from './MainNavigationLink.svelte'
+
+    export let onClose: () => void
+</script>
+
+<div class="root">
+    <div class="content" use:onClickOutside on:click-outside={onClose}>
+        <header>
+            <button class="close-button" on:click={onClose}>
+                <Icon svgPath={mdiClose} aria-label="Close sidebar navigation" />
+            </button>
+
+            <a href="/search" class="logo-link">
+                <SourcegraphLogo width="9.1rem" />
+            </a>
+        </header>
+
+        <nav>
+            <ul class="list">
+                {#each mainNavigation as entry (entry.label)}
+                    <li>
+                        <MainNavigationLink {entry} />
+                        {#if isNavigationMenu(entry) && entry.children.length > 0}
+                            <ul class="sub-list">
+                                {#each entry.children as subEntry (subEntry.label)}
+                                    <li aria-current={isCurrent(subEntry, $page) ? 'page' : 'false'}>
+                                        <MainNavigationLink entry={subEntry} />
+                                    </li>
+                                {/each}
+                            </ul>
+                        {/if}
+                    </li>
+                {/each}
+            </ul>
+        </nav>
+    </div>
+</div>
+
+<style lang="scss">
+    .root {
+        top: 0;
+        left: 0;
+        bottom: 0;
+        right: 0;
+        position: fixed;
+        z-index: 1;
+        // Fixed overlay color TODO: find a better design token for it
+        background-color: rgba(172, 182, 192, 0.2);
+
+        .content {
+            display: flex;
+            flex-direction: column;
+            width: 18.75rem;
+            height: 100%;
+            transform: unset;
+            border: none;
+            padding: 0;
+            background-color: var(--color-bg-1);
+        }
+
+        .close-button {
+            border: none;
+            padding: 0.35rem 0.35rem;
+            border-radius: var(--border-radius);
+            display: flex;
+            align-items: center;
+            background-color: transparent;
+
+            &:hover {
+                background-color: var(--secondary-2);
+            }
+
+            :global([data-icon]) {
+                width: 1rem;
+                height: 1rem;
+                color: var(--icon-color);
+            }
```

Comment:
`Icon` has a `--size` CSS variable that you can just set. Some for `color`. You don't have to use `global` here:

```suggestion
--size: 1rem;
--color: var(--icon-color)
```

We should rename those variables to be specific to the component though, e.g. `--icon-size`.
 
The same goes for all other places where you are targeting the `Icon` component.

Reply:
>Icon has a --size CSS variable that you can just set. Some for color. You don't have to use global here:

I don't think I see it in the Icon component (did you confuse it with Avatar component?), I can add it, though, if you think this would be better.

Reply:
Whoops, it accepts a size property.